Squeezer/src/main/java/uk/org/ngo/squeezer/service/SqueezeService.java

```diff
@@ -775,23 +822,149 @@ private void downloadSong(@NonNull Uri url, String title, @NonNull Uri serverUrl
                     .addRequestHeader("Authorization", "Basic " + base64EncodedCredentials);
             long downloadId = downloadManager.enqueue(request);
 
-            Util.crashlyticsLog("Registering new download");
-            Util.crashlyticsLog("downloadId: " + downloadId);
-            Util.crashlyticsLog("tempFile: " + tempFile);
-            Util.crashlyticsLog("localPath: " + localPath);
-
-            if (!downloadDatabase.registerDownload(downloadId, tempFile, localPath)) {
+            if (!downloadDatabase.registerDownload(downloadId, tempFile, localPath, albumArtUrl)) {
                 Util.crashlyticsLog(Log.WARN, TAG, "Could not register download entry for: " + downloadId);
                 downloadManager.remove(downloadId);
             }
         }
     }
 
+    @TargetApi(Build.VERSION_CODES.GINGERBREAD)
+    private void handleDownloadComplete(long id) {
+        final DownloadStorage downloadStorage = new DownloadStorage(this);
+        final DownloadDatabase downloadDatabase = new DownloadDatabase(this);
+        final DownloadManager downloadManager = (DownloadManager) getSystemService(DOWNLOAD_SERVICE);
+        final DownloadManager.Query query = new DownloadManager.Query().setFilterById(id);
+        final Cursor cursor = downloadManager.query(query);
+        try {
+            if (cursor.moveToNext()) {
```

Comment:
Maybe reduce the indentation levels here so the code's a bit more readable?

Something like:

```
if (!cursor.moveToNext()) {
    cursor.close();
    return;
}

...
final DownloadDatabase.DownloadEntry downloadEntry = downloadDatabase.popDownloadEntry(downloadId);
if (downloadEntry == null) {
    Util.crashlyticsLog(Log.ERROR, TAG, "Download database does not have an entry for " + format(status, reason, title, url, local_url));
    cursor.close()
    return;
}

if (status != DownloadManager.STATUS_SUCCESSFUL) {
    cursor.close();
    Util.crashlyticsLog(Log.ERROR, TAG, "Unsuccessful download " + format(status, reason, title, url, local_url));
    return;
}

try {
    File tempFile = ...
}